### ISSUE TITLE
Assertion failure when switching to station only mode

### ISSUE THREAD

**Issue by JimGaluska** [chmorgan/esphttpd-freertos#19]
There is an assertion failure on line 284 in file main/user_main.c when ever attempting to switch to standalone STA mode.  This issue is seen in git commit #ffffd5c.

**Instructions to reproduce:** 
1) On an ESP32, Checkout, compile, flash, and monitor version #ffffd5c.
2) Connect to the 'ESP' WiFi AP, browse to http://192.168.4.1, and select 'Configure your WIFI Network' option.
3) Click on the 'Go to AP+STA mode' button.  Wait for the screen to refresh.
4) Enter your SSID (or click the scan button and select your SSID), enter your password, and click on the 'Connect!' button.  After the screen has changed, click on the 'here' button to use the ESP's new IP address.
5) Re-select 'Configure your WIFI Network' option.
6) Click on the "Go to standalone STA mode' button, The esp will assert on line 284 in file main/user_main.c

**Additional Information:**
Modifying line 284 from:
 ```
ESP_ERROR_CHECK( esp_wifi_ap_get_sta_list(&sta_list));
```
to
 ```
 esp_wifi_ap_get_sta_list(&sta_list);
```

apparently makes the software work, but I am too new to this project to know if this is the correct way to fix this issue.

Also, the error return from function ```esp_wifi_ap_get_sta_list``` is ESP_ERR_WIFI_MODE (0x305) and
```sta_list.num``` is equal to 1 upon this functions return.

**Comment by phatpaul** [comment]
Yep, my bad.  Fixed in #20. Please test it.  Thanks for reporting.
-Paul

**Comment by kylix34** [comment]
I notice the same behaviour. After flashing the firmware, I can successfully connect to 192.168.4.1, associate the ESP to my WiFi network. All the pages work.

If I switch to STA Mode only and reboot, I can no longer ping the ESP.
